ElementX/Sources/Screens/RoomScreen/ComposerToolbar/CompletionSuggestionService.swift

```diff
@@ -83,23 +85,26 @@ final class CompletionSuggestionService: CompletionSuggestionServiceProtocol {
     
     // MARK: - Private
     
-    private func detectTriggerInText(_ text: String?) -> SuggestionTrigger? {
-        guard let text else {
-            return nil
-        }
-        
-        let components = text.components(separatedBy: .whitespaces)
+    private func detectTriggerInText(_ text: String, selectedRange: NSRange) -> SuggestionTrigger? {
+        let matches = text.matches(of: SuggestionTriggerRegex.at)
+        let match = matches
+            .filter { matchResult in
+                let lowerBound = matchResult.range.lowerBound.utf16Offset(in: matchResult.base)
+                let upperBound = matchResult.range.upperBound.utf16Offset(in: matchResult.base)
+                return selectedRange.location >= lowerBound
+                    && selectedRange.location <= upperBound
+                    && selectedRange.length <= upperBound - lowerBound
+            }
+            .first
```

Comment:
We can use [`first(where:)`](https://developer.apple.com/documentation/swift/array/first(where:)) here rather than filtering and then taking the first element.
```suggestion
            .first { matchResult in
                let lowerBound = matchResult.range.lowerBound.utf16Offset(in: matchResult.base)
                let upperBound = matchResult.range.upperBound.utf16Offset(in: matchResult.base)
                return selectedRange.location >= lowerBound
                    && selectedRange.location <= upperBound
                    && selectedRange.length <= upperBound - lowerBound
            }
```

Reply:
How do you know the problem, I am confused as to why the unit tests don't pass 😆.
I haven't notice this way to find result.